### ISSUE TITLE
RE: Compilation of the .elf file or is everything already done?

### ISSUE THREAD

**Issue by silver2row** [Teknic-ClearPath/TekTips-BeagleBone-Examples#1]
Hello,

I am trying number 9. of page five on the instructions for compiling the source to the BeagleBone Black. I have not been able to run any source yet on the BBB but I am still trying.

part f. ii. 

It shows that there is a compiled .elf file. Where is this file? I am not showing it on my browser in Eclipse after building. Is the .elf file the .so file?

Seth

P.S. I am trying to make your products work w/ the BBB as of now and I guess, as you can tell, I am stuck.

**Comment by silver2row** [comment]
Hello,

Okay. I understand now. .o, .so, core files, and executables are all part of .elf files. Okay. 

...

Is it possible to test the BBB w/out the hardware or do I need to have an operating Clearpath Servo system w/ Hub for servo exchange to have to test the source?

Seth

P.S. I am waiting on some cables now. I just wanted to have my BBB, SoC ready to have the compilation working for when these cables and non-switching power supply come in.

**Comment by Teknic-ClearPath** [comment]
Hi silver2row,

Sorry for the delayed response.  Generally the fastest way to get a technical question answered is to submit a contact request through our website (https://www.teknic.com/contact/) or simply calling into the main office.

As for your question, the most you could test without hardware is that the code compiles on the beaglebone and can execute without throwing any general systems errors.  Unfortunately there is no way to test functionality without hardware so any code you write will not be able to successfully open a port and so you should expect to see errors thrown from the library if you run without hardware.

Sincerely,
Brendan F, Teknic Servo Systems Engineer

**Comment by silver2row** [comment]
Hello Sir,

Yes sir, I will try to contact your company about this query.

Seth

P.S. That is all I was trying to do before I get more motors and the power supply.
